src/main/java/com/redhat/red/build/finder/BuildFinder.java

```diff
@@ -76,37 +76,44 @@ public BuildFinder(ClientSession session, BuildConfig config) {
             return Collections.emptyMap();
         }
 
-        Set<String> extensionsToCheck = new TreeSet<>();
         final Instant startTime = Instant.now();
+        List<String> archiveExtensions = config.getArchiveExtensions();
+        Set<String> extensionsToCheck = new TreeSet<>();
 
-        try {
-            Map<String, KojiArchiveType> allTypesMap = session.getArchiveTypeMap();
-            Set<String> allTypes = allTypesMap.values().stream().map(KojiArchiveType::getName).collect(Collectors.toSet());
+        Map<String, KojiArchiveType> allTypesMap;
 
-            LOGGER.debug("There are {} known Koji archive types: {}", allTypes.size(), allTypes);
+        try {
+            allTypesMap = session.getArchiveTypeMap();
+        } catch (KojiClientException e) {
+            LOGGER.error("Koji client error: {}", red(e.getMessage()));
+            session.close();
+            return Collections.emptyMap();
+        }
```

Comment:
We appear to be requesting the archive types even if the user has explicitly specified them? How about, if checksum-only is enabled it always uses the configuration archive types. Otherwise, if config archive types is specified it uses that, falling back to making the koji getArchiveTypeMap call? While this does move the call earlier its only used then in 'online' mode ?

Reply:
The problem with that is that the user can specify any extension to exclude from the checksums, but that extension is not necessarily a valid Koji archive type extension.

Reply:
It pretty much does this already. For the checksums, it uses the pre-configured archive extensions. For the builds, it ensures that the pre-configured archive extensions are valid (e.g., it takes the intersection of the specified types and all known types). This is why it still requests the types even when they are specified. This is just a single Koji API call, so it should not be expensive.